gc/base/Configuration.cpp

```diff
@@ -455,6 +456,49 @@ MM_Configuration::initializeGCThreadCount(MM_EnvironmentBase* env)
 	}
 }
 
+void
+MM_Configuration::initializeGCParameters(MM_EnvironmentBase* env)
+{
+	MM_GCExtensionsBase* extensions = env->getExtensions();
+	
+	/* TODO 108399: May need to adjust -Xmn*, -Xmo* values here if not fully specified on startup options */
+
+	/* initialize packet lock splitting factor */
+	if (0 == extensions->packetListSplit) {
+		if (16 >= extensions->gcThreadCount) {
+			extensions->packetListSplit = extensions->gcThreadCount;
+		} else if (32 >= extensions->gcThreadCount) {
+			extensions->packetListSplit = 16 + ((extensions->gcThreadCount - 16) / 4);
+		} else {
+			extensions->packetListSplit = 20 + ((extensions->gcThreadCount - 32) / 8);
+		}
+	}
+
+	/* initialize scan cache lock splitting factor */
+	if (0 == extensions->cacheListSplit) {
```

Comment:
The instance fields being initialized here are used in other components
(eg, marking scheme uses work packets, which depend on packetListSplit,
VLHGC uses scan caches, which depend on cacheListSplit, VLHGC uses
splitFreeListSpltAmount). I suggest leaving these open (not dependent on
OMR_GC_MODRON_SCAVENGER).



From:	Charlie Gracie <notifications@github.com>
To:	eclipse/omr <omr@noreply.github.com>
Cc:	Kim T Briggs/Ottawa/IBM@IBMCA
Date:	04/18/2016 08:15 AM
Subject:	Re: [eclipse/omr] FIX #51 : Merge generational GC (gencon) into
            OMR (#65)



In gc/base/Configuration.cpp:


> +
> +		 /* TODO 108399: May need to adjust -Xmn*, -Xmo* values here if not
fully specified on startup options */
> +
> +		 /* initialize packet lock splitting factor */
> +		 if (0 == extensions->packetListSplit) {
> +		 		 if (16 >= extensions->gcThreadCount) {
> +		 		 		 extensions->packetListSplit =
extensions->gcThreadCount;
> +		 		 } else if (32 >= extensions->gcThreadCount) {
> +		 		 		 extensions->packetListSplit = 16 +
((extensions->gcThreadCount - 16) / 4);
> +		 		 } else {
> +		 		 		 extensions->packetListSplit = 20 +
((extensions->gcThreadCount - 32) / 8);
> +		 		 }
> +		 }
> +
> +		 /* initialize scan cache lock splitting factor */
> +		 if (0 == extensions->cacheListSplit) {



this looks like scavenger specific code which should be under the compile
time flag


—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub